torch/csrc/cuda/Module.cpp

```diff
@@ -382,7 +383,28 @@ PyObject * THCPModule_initExtension(PyObject *self)
   if (!THCPModule_initCuda(torch_module)) {
     return NULL;
   }
-  Py_RETURN_NONE;
+
+  // Add class and method to torch.cuda
+  auto m = py::handle(torch_module).cast<py::module>();
+  py::class_<cudaDeviceProp>(m,"_CudaDeviceProperties")
+    .def_readonly("name", &cudaDeviceProp::name)
+    .def_readonly("major", &cudaDeviceProp::major)
+    .def_readonly("minor", &cudaDeviceProp::minor)
+    .def_readonly("is_multi_gpu_board", &cudaDeviceProp::isMultiGpuBoard)
+    .def_readonly("is_integrated", &cudaDeviceProp::integrated)
+    .def_readonly("multi_processor_count", &cudaDeviceProp::multiProcessorCount)
+    .def_readonly("total_memory", &cudaDeviceProp::totalGlobalMem)
+    .def("__repr__", [](const cudaDeviceProp &prop) {
+      char repr[256];
+      snprintf(repr, 256, "_CudaDeviceProperties(name=\"%s\", major=%d, minor=%d, total_memory=%dMB, multi_processor_count=%d)",
```

Comment:
oh there's a much simpler way:
```cpp
std::ostringstream stream;
stream << "_CudaDeviceProperties(name=" << prop.name << ", major=" << prop.major
       << ", minor=" << prop.minor << ", total_memory=" << prop.totalGlobalMem / (1024 * 1024)
       << ", multi_processor_count=" << prop.multiProcessorCount << ")";
```

Reply:
ok, wasn't sure if we wanted to include `stringstream` only for that. will change it.

ah, I wish C++ had `"{}".format()`... I prefer keeping the format and the data apart!